src/components/review/ReviewConfiguration.jsx

```diff
@@ -58,8 +62,86 @@ const ReviewDescriptionList = ({ children }) => {
     );
 };
 
-export const ReviewConfiguration = ({ deviceData, diskSelection, language, localizationData, requests, idPrefix, setIsFormValid, storageScenarioId, accounts }) => {
+const ChangeHostname = ({ initHostname }) => {
+    const [currentHostname, setCurrentHostname] = useState("");
+    const [error, setError] = useState([]);
+    const [isModalOpen, setIsModalOpen] = React.useState(false);
+
+    const validateHostname = (value) => {
+        const error = [];
+        if (value.length > 64) { error.push(_("Real host name must be 64 characters or less")) }
+        if (value.match(/[.a-z0-9-]*/)[0] !== value || value.indexOf("..") !== -1) { error.push(_("Real host name can only contain lower-case characters, digits, dashes, and periods (with populated subdomains)")) }
+
+        setError(error);
+    };
+
+    const onHostnameChanged = (value) => {
+        validateHostname(value);
+        setCurrentHostname(value);
+    };
+    const onSubmit = (event) => {
+        setHostname({ hostname: currentHostname })
+                .then(() => {
+                    handleModalToggle();
+                })
+                .catch(ex => {
+                    console.info({ ex });
+                });
+
+        if (event) { event.preventDefault() }
+        return false;
+    };
+
+    const handleModalToggle = _event => {
+        setIsModalOpen(!isModalOpen);
+    };
+
+    const disabled = error.length || (initHostname === currentHostname);
+    return (
+        <>
+            <Button
+              id="system_information_hostname_button" variant="link"
+              onClick={handleModalToggle}
+              isInline aria-label="edit hostname">
+                {_("edit")}
+            </Button>
+            <Modal
+              footer={
+                  <>
+                      <Button variant="primary" isDisabled={disabled} onClick={onSubmit}>{_("Change")}</Button>
+                      <Button variant="link" onClick={handleModalToggle}>{_("Cancel")}</Button>
+                  </>
+              }
+              id="system_information_change_hostname"
+              isOpen={isModalOpen}
+              onClose={handleModalToggle}
+              position="top"
+              title={_("Change host name")}
+              variant="small"
+            >
+                <Form isHorizontal onSubmit={() => onSubmit}>
```

Comment:
onSubmit expects a function. You can either use onSubmit={onSubmit} or use onSubmit={() => onSubmit()} - notice () after function name.

Please use the former for consistency.